src/main/java/com/epam/rd/autotasks/springemployeecatalog/extractor/SimpleEmployeeExtractor.java

```diff
@@ -0,0 +1,95 @@
+        Long id = resultSet.getLong(ID.getColumnName() + SUFFIX_E);
+        String firstName = resultSet.getString(FIRSTNAME.getColumnName() + SUFFIX_E);
+        String lastName = resultSet.getString(LASTNAME.getColumnName() + SUFFIX_E);
+        String middleName = resultSet.getString(MIDDLENAME.getColumnName() + SUFFIX_E);
+        FullName fullName = new FullName(firstName, lastName, middleName);
+        Position position = Position.valueOf(resultSet.getString(POSITION.getColumnName() + SUFFIX_E));
+        LocalDate hired = resultSet.getDate(HIREDATE.getColumnName() + SUFFIX_E).toLocalDate();
+        BigDecimal salary = resultSet.getBigDecimal(SALARY.getColumnName() + SUFFIX_E);
+        long departmentId = resultSet.getLong(ID.getColumnName() + SUFFIX_D_E);
+        String name = resultSet.getString(NAME.getColumnName() + SUFFIX_D_E);
+        String location = resultSet.getString(LOCATION.getColumnName() + SUFFIX_D_E);
+        Department department = departmentId != 0L ? new Department(departmentId, name, location) : null;
+        Long managerId = resultSet.getLong(MANAGER.getColumnName() + SUFFIX_E);
+
+        Employee manager = null;
+
+        if (managerId != 0L) {
+            String managerFirstname = resultSet.getString(FIRSTNAME.getColumnName() + SUFFIX_M);
+            String managerLastname = resultSet.getString(LASTNAME.getColumnName() + SUFFIX_M);
+            String managerMiddlename = resultSet.getString(MIDDLENAME.getColumnName() + SUFFIX_M);
+            FullName managerFullName = new FullName(managerFirstname, managerLastname, managerMiddlename);
+            Position managerPosition = Position.valueOf(resultSet.getString(POSITION.getColumnName() + SUFFIX_M));
+            LocalDate managerHired = resultSet.getDate(HIREDATE.getColumnName() + SUFFIX_M).toLocalDate();
+            BigDecimal managerSalary = resultSet.getBigDecimal(SALARY.getColumnName() + SUFFIX_M);
+            long managerDepartmentId = resultSet.getLong(ID.getColumnName() + SUFFIX_D_M);
+            String managerDepartmentName = resultSet.getString(NAME.getColumnName() + SUFFIX_D_M);
+            String managerDepartmentLocation = resultSet.getString(LOCATION.getColumnName() + SUFFIX_D_M);
+            Department managerDepartment = managerDepartmentId != 0L ?
+                    new Department(managerDepartmentId, managerDepartmentName, managerDepartmentLocation) : null;
+
+            manager = new Employee(
+                    managerId,
+                    managerFullName,
+                    managerPosition,
+                    managerHired,
+                    managerSalary,
+                    null,
+                    managerDepartment);
+        }
+
+        return new Employee(
+                id,
+                fullName,
+                position,
+                hired,
+                salary,
+                manager,
+                department);
+    }
```

Comment:
It could be refactored to following way. Small methods are better than big.
`private static final String SUFFIX_D_E = "_d_e";
    private static final String SUFFIX_D_M = "_d_m";
    private static final String SUFFIX_E = "_e";
    private static final String SUFFIX_M = "_m";

    @Override
    public List<Employee> extractData(ResultSet resultSet) throws SQLException {
        List<Employee> employees = new ArrayList<>();
        while (resultSet.next()) {
            employees.add(createEmployeeWithManager(resultSet));
        }
        return employees;
    }

    public static Employee createEmployeeWithManager(ResultSet resultSet) throws SQLException {

        Long managerId = resultSet.getLong(getColumnName(MANAGER, SUFFIX_E));
        Optional<Employee> manager = getDirectManager(resultSet, managerId);

        Long id = resultSet.getLong(getColumnName(ID, SUFFIX_E));
        return getEmployee(resultSet, id, manager.orElse(null), SUFFIX_E, SUFFIX_D_E);
    }

    private static Optional<Employee> getDirectManager(ResultSet resultSet, Long managerId) throws SQLException {
        if (isManagerExists(managerId)) {
            return Optional.of(getEmployee(resultSet, managerId, null, SUFFIX_M, SUFFIX_D_M));
        }
        return Optional.empty();
    }

    private static Employee getEmployee(ResultSet resultSet, Long id, Employee directManager, String suffix, String departmentSuffix) throws SQLException {
        FullName managerFullName = getFullName(resultSet, suffix);
        Position managerPosition = Position.valueOf(resultSet.getString(getColumnName(POSITION, suffix)));
        LocalDate managerHired = resultSet.getDate(getColumnName(HIREDATE, suffix)).toLocalDate();
        BigDecimal managerSalary = resultSet.getBigDecimal(SALARY.getColumnName() + suffix);
        Optional<Department> department = getDepartment(resultSet, departmentSuffix);

        return new Employee(
                id,
                managerFullName,
                managerPosition,
                managerHired,
                managerSalary,
                directManager,
                department.orElse(null));
    }

    private static String getColumnName(TableColumnNamesEnum column, String suffix) {
        return column.getColumnName() + suffix;
    }

    private static Optional<Department> getDepartment(ResultSet resultSet, String departmentSuffix) throws SQLException {
        long managerDepartmentId = resultSet.getLong(getColumnName(ID, departmentSuffix));
        if (managerDepartmentId != 0L) {
            String managerDepartmentName = resultSet.getString(getColumnName(NAME,departmentSuffix));
            String managerDepartmentLocation = resultSet.getString(getColumnName(LOCATION, departmentSuffix));
            return Optional.of(new Department(managerDepartmentId, managerDepartmentName, managerDepartmentLocation));
        }
        return Optional.empty();
    }

    private static boolean isManagerExists(Long managerId) {
        return managerId != 0L;
    }

    private static FullName getFullName(ResultSet resultSet, String suffix) throws SQLException {
        String managerFirstname = resultSet.getString(FIRSTNAME.getColumnName() + suffix);
        String managerLastname = resultSet.getString(LASTNAME.getColumnName() + suffix);
        String managerMiddlename = resultSet.getString(MIDDLENAME.getColumnName() + suffix);
        return new FullName(managerFirstname, managerLastname, managerMiddlename);
    }`